### PR TITLE
Add tests for clip function with None parameter to min/max

### PR DESCRIPTION
**Pull request by take-cheeze**
Spin-off from: #7285 

Also adds some `None` checks too.

### CODE REVIEW

File: chainer/functions/math/clip.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ class Clip(function_node.FunctionNode):
 
     def __init__(self, x_min, x_max):
```
Comment (hvy) [inline chainer/functions/math/clip.py]: Could you also check that not both of min and max are `None`, and if so, throw an appropriate error.

File: chainer/functions/math/clip.py [hunk 2]
```diff
@@ -38,7 +38,11 @@ def backward(self, indexes, grad_outputs):
 class ClipGrad(function_node.FunctionNode):
 
     def __init__(self, x, x_min, x_max):
-        self.cond = (x_min <= x) * (x <= x_max)
+        self.cond = True
```
Comment (hvy) [inline chainer/functions/math/clip.py]: How about asserting above this line that not both of min and max are `None` since it's an assumption that this class makes.

File: chainer/functions/math/clip.py [hunk 3]
```diff
@@ -10,8 +10,11 @@ class Clip(function_node.FunctionNode):
     """Clips (limits) elements of input variable."""
 
     def __init__(self, x_min, x_max):
+        if (x_min is None) and (x_max is None):
```
Comment (hvy) [inline chainer/functions/math/clip.py]: Thanks for adding the check. It's a detail but you can skip the parenthesis.
```suggestion
        if x_min is None and x_max is None:
```

File: chainer/functions/math/clip.py [hunk 4]
```diff
@@ -10,8 +10,11 @@ class Clip(function_node.FunctionNode):
...
+            raise ValueError('must set either max or min')
+
         # x_min must be less than x_max.
-        if x_min >= x_max:
+        if (x_min is not None) and (x_max is not None) and (x_min >= x_max):
```
Comment (hvy) [inline chainer/functions/math/clip.py]: I think you can skip the parenthesis here too. But maybe this is more readable. What do you think?

File: chainer/functions/math/clip.py [hunk 5]
```diff
@@ -38,7 +41,14 @@ def backward(self, indexes, grad_outputs):
 class ClipGrad(function_node.FunctionNode):
 
     def __init__(self, x, x_min, x_max):
-        self.cond = (x_min <= x) * (x <= x_max)
+        if (x_min is None) and (x_max is None):
```
Comment (hvy) [inline chainer/functions/math/clip.py]: Again, I think it's cleaner to skip parenthesis.

File: tests/chainer_tests/functions_tests/math_tests/test_clip.py [hunk 6]
```diff
@@ -8,6 +8,7 @@
 from chainer import gradient_check
 from chainer import testing
 from chainer.testing import attr
+from chainer.functions.math.clip import (Clip, ClipGrad)
```
Comment (hvy) [inline tests/chainer_tests/functions_tests/math_tests/test_clip.py]: Actually could you check the input validation using the functions `chainer.functions.clip` and not the `FunctionNode`s? The `FunctionNode`s are sort of "private" although this is not clearly communicated. That's also why I suggested asserting in `ClipGrad`. Another few tips general tips is that we don't import non-modules and we only have single imports per line 🙂

File: chainer/functions/math/clip.py (hunk 4 above)
Comment (take-cheeze) [reply chainer/functions/math/clip.py]: I think `None` checks should be separate `if` statement for readability.
Is that OK in Chainer Python style?

Comment (hvy) [reply chainer/functions/math/clip.py]: That's fine too.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM, just let me rewrite the description to more reflect the actual changes introduced.
